### ISSUE TITLE
`jest/no-done-callback` how to use `done.fail()` in `beforeEach` with a `Promise`?

### ISSUE THREAD

**Issue by martinfrancois** [jest-community/eslint-plugin-jest#698]
When using a callback with `done()`, it will trigger the `jest/no-done-callback` rule, but how can we use `done.fail()` when using Promises in `beforeEach`?

In a similar question: https://github.com/jest-community/eslint-plugin-jest/issues/657 it was about tests, in which it was suggested to use `throw new Error`, but this doesn't work for `beforeEach`.

The concrete use case for this is the following: We want a test to fail when an error is logged to the console. Previously we did this using the following code:
```ts
export const failOnConsoleError = (done: jest.DoneCallback): void => {
  jest.spyOn(global.console, 'error').mockImplementation((e) => {
    done.fail(e);
  });
  done();
};

beforeEach((done) => {
    failOnConsoleError(done);
});
```

As far as I can see, there's no way to do the same with a `Promise`. If that's the case, we should add an exclusion for those cases. But I'm more than glad if anyone knows a way to do it anyway.

**Comment by SimenB** [comment]
Your old implementation has some weird characteristics where you say "I'm done" multiple times.

Regardless, something like this should work.

```ts
let consoleErrorSpy;

beforeEach(() => {
  consoleErrorSpy = jest.spyOn(console, 'error');
});

afterEach(() => {
  // Fail tests on any warning
  expect(consoleErrorSpy).not.toHaveBeenCalled();

  consoleErrorSpy.mockRestore();
});
```

You can also choose to just `throw new Error('boom')` inside of your mock implementation

**Comment by martinfrancois** [comment]
@SimenB the approach you proposed does indeed work, but it produces a different eslint rule violation:
```
error    Expect must be inside of a test block  jest/no-standalone-expect
```

I also already tried throwing an error in the mock implementation, as you mentioned, but that doesn't work - when an error is logged to the `console`, it will not fail the test:
```
beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {
      throw new Error();
    });
});
```
